ci/test-clang-docker.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+set -eu
+
+if [[ ! -d /repo ]]; then
+    echo "[-] This script needs to be run within Docker! Use test-clang.sh instead."
+    exit 1
+fi
+
+CC=/repo/toolchain/bin/clang
+CXX=/repo/toolchain/bin/clang++
+hello_c=/repo/llvm-project/hello.c
+hello_cpp=/repo/llvm-project/hello.cpp
+hello_exe=/tmp/hello
+host_arch=$(uname -m)
+
+# Print clang information, which makes sure it can run
+echo "[+] Testing 'clang --version'"
+"$CC" --version
+
+case "$(source /usr/lib/os-release; echo "$ID")" in
+    arch)
+        musl_cc_flags=(
+            -B /usr/lib/musl/lib     # Scrt1.o, crti.o, crtn.o
+            -I /usr/lib/musl/include # stdio.h
+            -L /usr/lib/musl/lib     # -lc
+        )
+
+        echo "[+] Updating OS and installing musl"
+        pacman -Syyu --noconfirm musl
+        ;;
+
+    fedora)
+        musl_cc_flags=(
+            -B /usr/"$host_arch"-linux-musl/lib64   # Scrt1.o, crti.o, crtn.o
+            -I /usr/"$host_arch"-linux-musl/include # stdio.h
+            -L /usr/"$host_arch"-linux-musl/lib64   # -lc
```

Comment:
Sure, I can test that.

Reply:
Specifying just `--target` doesn't work:

```
[+] Testing dynamically linking hello.c against musl ('-B, -I, -L')
hello world
[+] Testing statically linking hello.c against musl ('-B, -I, -L')
hello world
[+] Testing dynamically linking hello.c against musl ('--target')
/repo/llvm-project/hello.c:1:10: fatal error: 'stdio.h' file not found
#include <stdio.h>
         ^~~~~~~~~
1 error generated.
```

```diff
diff --git a/ci/test-clang-docker.sh b/ci/test-clang-docker.sh
index 3a492ed..54c1c3a 100755
--- a/ci/test-clang-docker.sh
+++ b/ci/test-clang-docker.sh
@@ -52,14 +52,22 @@ esac
 # MUSL #
 ########

-echo "[+] Testing dynamically linking hello.c against musl"
+echo "[+] Testing dynamically linking hello.c against musl ('-B, -I, -L')"
 "$CC" "${musl_cc_flags[@]}" -o "$hello_exe" "$hello_c"
 "$hello_exe"

-echo "[+] Testing statically linking hello.c against musl"
+echo "[+] Testing statically linking hello.c against musl ('-B, -I, -L')"
 "$CC" "${musl_cc_flags[@]}" -static -o "$hello_exe" "$hello_c"
 "$hello_exe"

+echo "[+] Testing dynamically linking hello.c against musl ('--target')"
+"$CC" --target="$host_arch"-linux-musl -o "$hello_exe" "$hello_c"
+"$hello_exe"
+
+echo "[+] Testing statically linking hello.c against musl ('--target')"
+"$CC" --target="$host_arch"-linux-musl -static -o "$hello_exe" "$hello_c"
+"$hello_exe"
+
 echo "[+] Testing dynamically linking hello.cpp against musl ('--rpath')"
 # --rpath to avoid having to pull in libc++ and libunwind from distribution
 "$CXX" "${musl_cc_flags[@]}" -Wl,--rpath=/repo/toolchain/lib/"$host_arch"-alpine-linux-musl -o "$hello_exe" "$hello_cpp"
@@ -69,6 +77,10 @@ echo "[+] Testing dynamically linking hello.cpp against musl ('LD_LIBRARY_PATH')
 "$CXX" "${musl_cc_flags[@]}" -o "$hello_exe" "$hello_cpp"
 LD_LIBRARY_PATH=/lib:/repo/toolchain/lib/"$host_arch"-alpine-linux-musl "$hello_exe"

-echo "[+] Testing statically linking hello.cpp against musl"
+echo "[+] Testing statically linking hello.cpp against musl ('-B, -I, -L')"
 "$CXX" "${musl_cc_flags[@]}" -static -lc++abi -o "$hello_exe" "$hello_cpp"
 "$hello_exe"
+
+echo "[+] Testing statically linking hello.cpp against musl ('--target')"
+"$CXX" --target="$host_arch"-linux-gnu -static -lc++abi -o "$hello_exe" "$hello_cpp"
+"$hello_exe"
```

Reply:
Ok then, I'll need to spend some time with `-v` to see why the search paths aren't what I had expected. This is fine to add for now.